src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -1256,11 +1261,56 @@ void BluezObjectsCleanup(BluezEndpoint * apEndpoint)
     EndpointCleanup(apEndpoint);
 }
 
+#if !CHIP_BYPASS_ADDITIONAL_DATA_ADVERTISING
+static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * characteristic)
+{
+    // Construct the TLV for the additional data
+    GVariant * cValue = nullptr;
+    CHIP_ERROR err    = CHIP_NO_ERROR;
+    TLVWriter writer;
+    chip::System::PacketBufferHandle bufferHandle = chip::System::PacketBuffer::New();
+    chip::System::PacketBuffer * buffer           = bufferHandle.Get_ForNow();
+
+    if (characteristic == nullptr)
+    {
+        return;
+    }
+
+    writer.Init(buffer);
+    TLVType containerType;
+    err = writer.StartContainer(AnonymousTag, kTLVType_Structure, containerType);
+    SuccessOrExit(err);
+
+    // Adding the rotating device id to the TLV data
+    err = writer.PutString(CommonTag(0), CHIP_ROTATING_DEVICE_ID);
```

Comment:
This also does not match the spec, insofar as the spec says anything.  I filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/677 on the lack of spec clarity, but it seems to be trying to define that these are context-specific tags.  So this should be `ContextTag`, I would think.